ethcore/src/client/client.rs

```diff
@@ -122,11 +122,13 @@ impl SleepState {
 /// Call `import_block()` to import a block asynchronously; `flush_queue()` flushes the queue.
 pub struct Client {
 	mode: Mode,
-	chain: Arc<BlockChain>,
-	tracedb: Arc<TraceDB<BlockChain>>,
+	chain: RwLock<Arc<BlockChain>>,
```

Comment:
Ah, ok. I'm sure that's not happening cause this pointer is not exposed out of the client. It is only used in TraceDB and we should really just pass a reference there as @rphmeier suggested.